src/Evaluator.ml

```diff
@@ -86,6 +100,20 @@ let rec eval_open gamma Location.{ data = expr; _ } =
       match pv with
       | Val.Pair { v1 = _; v2 } -> return v2
       | _ -> Result.fail @@ `EvaluationError "snd is stuck" )
+  | Nat { n } -> return @@ Val.Nat { n }
+  | BinOp { op; e1; e2 } -> (
+      let%bind lhs = eval_open gamma e1 in
+      let%bind rhs = eval_open gamma e2 in
+      match (lhs, rhs) with
+      | Val.Nat { n = n1 }, Val.Nat { n = n2 } -> (
+          match op with
+          | Add -> return @@ Val.Nat { n = Nat.add n1 n2 }
+          | Sub -> return @@ Val.Nat { n = Nat.sub n1 n2 }
+          | Mul -> return @@ Val.Nat { n = Nat.mul n1 n2 }
+          | Div -> return @@ Val.Nat { n = Nat.div n1 n2 } )
```

Comment:
We need to treat the case of division by zero here. Essentially catch the exception and turn it into `Error`.